WordPress/src/main/java/org/wordpress/android/ui/sitecreation/sitename/SiteCreationSiteNameViewModel.kt

```diff
@@ -38,28 +37,24 @@ class SiteCreationSiteNameViewModel @Inject constructor(
 
     fun start() {
         if (isInitialized) return
-        // TODO: analyticsTracker.trackSiteNameViewed()
-        Log.d("siteNameWip", "Site name viewed")
+        analyticsTracker.trackSiteNameViewed()
         isInitialized = true
     }
 
     fun onSkipPressed() {
-        // TODO: analyticsTracker.trackSiteNameSkipped()
-        Log.d("siteNameWip", "Site name skipped")
+        analyticsTracker.trackSiteNameSkipped()
         _onSkipButtonPressed.call()
     }
 
     fun onBackPressed() {
-        // TODO: analyticsTracker.trackSiteNameCancelled()
-        Log.d("siteNameWip", "Site name cancelled")
+        analyticsTracker.trackSiteNameCanceled()
         _onBackButtonPressed.call()
     }
 
     fun onSiteNameEntered() {
-        // TODO: analyticsTracker.trackSiteNameEntered(siteName?) maybe we don't want to track the names here?
         uiState.value?.siteName.let {
```

Comment:
This LGTM as is but I would suggest a [small change to align with iOS](https://github.com/wordpress-mobile/WordPress-iOS/pull/18366/files#diff-d5e86f4583cfcd36951f9e02271b24a63baad47c95f1f2cbfef393844edd0c7dR22)
```
        uiState.value?.siteName?.let {
            if (it.isBlank()) return
            analyticsTracker.trackSiteNameEntered(it)
            _onSiteNameEntered.value = it
        }
```
This would allow us to [make the function parameter non-null too](https://github.com/wordpress-mobile/WordPress-Android/pull/16330/files#diff-0e437bceed89370701ab725a4b9dd9e51914fadffd32be82979744f4f45cbdc1R292). Wdyt?

Reply:
> This would allow us to [make the function parameter non-null too](https://github.com/wordpress-mobile/WordPress-Android/pull/16330/files#diff-0e437bceed89370701ab725a4b9dd9e51914fadffd32be82979744f4f45cbdc1R292). Wdyt?

I love this suggestion! Will implement it 👍

Reply:
Applied the suggestion in https://github.com/wordpress-mobile/WordPress-Android/pull/16330/commits/58b05e50f6581f7ad79f24c2584e9473ac7a1593 & added a test for it in https://github.com/wordpress-mobile/WordPress-Android/pull/16330/commits/85727e2acad6b66334f7f535d64916d40df4f325.

Thank you for checking how this is in the iOS code 👍

Reply:
Thank you for the changes @ovitrif 🙇

Reply:
Ah, when I added this, I originally meant to do:

```kotlin
        uiState.value?.siteName?.let {
            analyticsTracker.trackSiteNameEntered(it)
            if (it.isBlank()) return
            _onSiteNameEntered.value = it
        }
```

but forgot to use the second optional chaining before the `let` (hence `isBlankOrNull` instead of just `isBlank`). This would have tracked blanks, but it's probably not useful, since it shouldn't be possible - and also, I think we could get this info anyway if needed. Thanks for the improvement :+1: